asllib/error.ml

```diff
@@ -317,6 +334,13 @@ module PPrint = struct
           "ASL Typing error:@ explicit@ parameter@ %S@ does@ not@ have@ a@ \
            corresponding@ defining@ argument."
           s
+    | BadParameterDecl (name, expected, actual) ->
+        fprintf f
+          "ASL Typing error: incorrect parameter declaration for %S, expected \
+           %S but %S provided"
+          name
+          ("{" ^ String.concat ", " expected ^ "}")
+          ("{" ^ String.concat ", " actual ^ "}")
```

Comment:
My intention here was to make the "correct" declaration copy-pastable (braces and all) - so I'll incorporate your suggestion without the `pp_quote_string`, as (if I understand correctly) this will introduce double-quotes around each identifier